src/btree/bt_cursor.c

```diff
@@ -708,8 +701,15 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
      */
     if (valid) {
         exact = cbt->compare;
-        ret = __cursor_kv_return(cbt, cbt->upd_value);
```

Comment:
What would happen to ret in this scenario previously?

Reply:
In the previous scenario if ret was set to 0, it would immediately go into setting exactp to exact and if HAVE_DIAGNOSTIC was enabled, would call this function `__wt_cursor_key_order_init`.
Otherwise if ret wasn't set to 0, we perform a `cursor_reset()` and restore the state of the cursor through `__cursor_state_restore()`